services/requester/single_tx_pool.go

```diff
@@ -177,53 +176,33 @@ func (t *SingleTxPool) buildTransaction(
 		}
 	}
 
-	// building and signing transactions should be blocking,
-	// so we don't have keys conflict
-	t.mux.Lock()
-	defer t.mux.Unlock()
+	accKey, err := t.fetchSigningAccountKey()
+	if err != nil {
+		return nil, err
+	}
 
-	accKey, err := t.keystore.Take()
+	coaAddress := t.config.COAAddress
+	accountKey, err := t.client.GetAccountKeyAtLatestBlock(ctx, coaAddress, accKey.Index)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := accKey.SetProposerPayerAndSign(flowTx, account); err != nil {
+	if err := accKey.SetProposerPayerAndSign(flowTx, coaAddress, accountKey); err != nil {
 		accKey.Done()
 		return nil, err
 	}
 
 	// now that the transaction is prepared, store the transaction's metadata
 	accKey.SetLockMetadata(flowTx.ID(), latestBlock.Height)
 
-	t.collector.OperatorBalance(account)
```

Comment:
I have removed the `OperatorBalance` metric, because now we use `GetAccountKeyAtLatestBlock`, and we don't have access to the COA balance.
I am planning to add this back, on this PR: https://github.com/onflow/flow-evm-gateway/pull/896. Generally, we don't have to emit this metric on every tx submission, we can do it at a less frequent rate.